core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java

```diff
@@ -314,10 +329,19 @@ private void completeLog(UfsJournalFile currentLog, long nextSequenceNumber) thr
       }
       return;
     }
-    LOG.info("Completing log {} with next sequence number {}", current, nextSequenceNumber);
     String completed = UfsJournalFile
         .encodeLogFileLocation(mJournal, currentLog.getStart(), nextSequenceNumber).toString();
 
+    try {
+      // Check again before the rename
+      checkIsWritable();
+    } catch (JournalClosedException e) {
+      // Do not throw error, just ignore if the journal is not writable
+      LOG.warn("Skipping completeLog() since journal is not writable. error: {}", e.getMessage());
+      return;
+    }
```

Comment:
We could still lose primacy after the check, but that's fine because the rename will not pick up the wrong journal file? (ie. we check for primacy between updating `current` and attempting rename)

Reply:
yeah, I was thinking about that. I don't know how else to prevent the gap, but this basically the smallest we can make this gap. I'm not sure what will happen if the failover happens after the check, but before the rename...

Reply:
I think it should be okay because it still is referencing the old journal file.